### PR TITLE
Traffic lights

### PR DESCRIPTION
**Pull request by PawelZ4krzewski**

### CODE REVIEW

File: core/src/com/put/urbantraffic/TrafficLight.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.put.urbantraffic;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+
+@Data
+@AllArgsConstructor
```
Comment (kezc) [inline core/src/com/put/urbantraffic/TrafficLight.java:8]: Doesn't `@Data` include `@AllArgConstructor`? I think it's redundant

File: core/src/com/put/urbantraffic/Crossing.java [hunk 2]
```diff
@@ -1,14 +1,252 @@
...
+//        if(y > car.getCarPosition().getY()){
+//            if(trafficLightsSupervisor.getBottomTrafficLight().getCurrentColor() == Light.GREEN){
+//                bottomCardField = car;
+//                return true;
+//            }
+//            return false;
+//        }
+//
+//        return false;
+//    }
```
Comment (kezc) [inline core/src/com/put/urbantraffic/Crossing.java]: :|

File: core/src/com/put/urbantraffic/Crossing.java [hunk 3]
```diff
@@ -1,14 +1,252 @@
...
+        Car.Way way = car.getWay();
+        Direction direction = car.getDirection();
+        System.out.println("Car " + car.getCarPosition() + " crossing " + x + " "+ y);
+        System.out.println(" way " + way + " direction "+ direction);
+        System.out.println("LightSupervisor: " + trafficLightsSupervisor.toString());
+        if (way == Car.Way.TOP) {
+            System.out.println("Traffic Light" + trafficLightsSupervisor.getBottomTrafficLight().getCurrentColor());
+            if(trafficLightsSupervisor.getBottomTrafficLight().getCurrentColor() == Light.GREEN){
+                if (direction == Direction.FORWARD) {
+                    if (forwardBottomCardField.size() == 0 && turnLeftBottomCardField.size() < 2) {
```
Comment (kezc) [inline core/src/com/put/urbantraffic/Crossing.java:95]: maybe we could merge outer if with nested one

File: core/src/com/put/urbantraffic/GlobalTime.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import lombok.Data;
+
+@Data
+public class GlobalTime {
+    int hour;
+    int minutes;
+    int seconds;
+    int miliseconds;
+
+    public GlobalTime(){
```
Comment (kezc) [inline core/src/com/put/urbantraffic/GlobalTime.java:12]: If we don't need a constructor generated by `@Data` maybe we could use a different annotation? e.g. `@Getter`, `@Setter`

File: core/src/com/put/urbantraffic/Crossing.java [hunk 5]
```diff
@@ -1,14 +1,252 @@
...
+
+        switch (car.getWay()) {
+            case TOP:
+                if(turnLeftBottomCardField.contains(car)) {
+                    turnLeftBottomCardField.remove(car);
+                }
+                else if(turnRightBottomCardField.contains(car)) {
+                    turnRightBottomCardField.remove(car);
+                }
+                else if(forwardBottomCardField.contains(car)) {
```
Comment (kezc) [inline core/src/com/put/urbantraffic/Crossing.java]: IntelliJ gives a warning here and in a couple of other cases below for `Unnecessary 'contains()' check `. Is it true? If not, can we suppress it?

**Comment by kezc** [comment]
The project doesn't compile

File: core/src/com/put/urbantraffic/TrafficLight.java (hunk 1 above)
Comment (PawelZ4krzewski) [reply core/src/com/put/urbantraffic/TrafficLight.java:8]: Without @AllArgConstructor City class throws errors. It's important to correct work.

**Review by kezc — APPROVED** [review]

**Pull request merged**